### PR TITLE
auto-updater: respond to Windows Service events

### PR DESCRIPTION
**Pull request by mniewrzal**
What: Implement ServiceMain in auto-updater binary, so it can be started and stopped as a Windows Service.

Why: We want to install the auto-updater as a Windows Service on Windows OS. See https://storjlabs.atlassian.net/browse/V3-2481

Please describe the tests: N/A

Please describe the performance impact: N/A

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: cmd/auto-updater/windows.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// Implements support for running the storage node as a Windows Service.
```
Comment (kaloyan-raev) [inline cmd/auto-updater/windows.go]: This comment needs an update: "storage node" --> "auto-updater for storage nodes".

File: cmd/auto-updater/windows.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				// Testing deadlock from https://code.google.com/p/winsvc/issues/detail?id=4
+				time.Sleep(100 * time.Millisecond)
+				changes <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				log.Println("Stop/Shutdown request received.")
+				changes <- svc.Status{State: svc.StopPending}
+
+				cancel()
+				// Sleep some time to give chance for goroutines finish cleanup after cancelling the context
+				time.Sleep(15 * time.Second)
```
Comment (kaloyan-raev) [inline cmd/auto-updater/windows.go]: I think we can have a shorter sleep for the auto-updater, e.g. 2-3 seconds. There shouldn't be many goroutines running, right?

Comment (mniewrzal) [reply cmd/auto-updater/windows.go]: I was thinking about it and yes, from long-running things we should have only loop and http requests.

File: cmd/auto-updater/windows.go (hunk 1 above)
Comment (mniewrzal) [reply cmd/auto-updater/windows.go]: ah missed that one, thanks

**Pull request merged**